core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java

```diff
@@ -477,4 +476,22 @@ public void testCreateShrinkIndexFails() throws Exception {
         ensureGreen();
         assertHitCount(client().prepareSearch("target").setSize(100).setQuery(new TermsQueryBuilder("foo", "bar")).get(), 20);
     }
+
+    public void testCreateIndexWaitForShardAllocation() throws Exception {
+        final int numDataNodes = randomIntBetween(1, 5);
+        final String indexName = "test-idx";
+        internalCluster().ensureAtLeastNumDataNodes(numDataNodes);
+        final Settings settings = Settings.builder()
+                                          .put(indexSettings())
+                                          .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(2, 7))
+                                          .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), randomIntBetween(0, numDataNodes - 1))
+                                          .build();
+        final CreateIndexResponse response = prepareCreate(indexName).setSettings(settings).setTimeout("10s").get();
+        assertTrue(response.isAcknowledged());
+        final int numDocs = 20;
+        for (int i = 0; i < numDocs; i++) {
+            client().prepareIndex(indexName, "type").setSource("{\"foo\" : \"bar\", \"i\" : " + i + "}").get();
+        }
+        assertThat(client().prepareSearch("target").setSize(numDocs).get().getHits().getTotalHits(), equalTo(numDocs));
```

Comment:
I think there is an `assertHitCount` you can use. Do you want `setSize(0)` as well?